### DISCUSSION TITLE
Multiple releases using the same, but configurable, Angular Standalone Application build

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#310] [Announcements]
# Multiple releases using the same, but configurable, Angular Standalone Application build

Configuring an Angular standalone application to be flexible enough to be released in different environments.

https://timdeschryver.dev/blog/multiple-releases-using-the-same-but-configurable-angular-standalone-application-build

**Comment by markgoho** [comment]
I think you linked to the old version of ApplicationConfig, here's the new one: https://angular.dev/api/core/ApplicationConfig

**Reply by timdeschryver** [reply]
Thanks @markgoho , it seems you're right!

**Comment by apryk** [comment]
Is there a way to inject this configuration object using standalone component "inject()" or the only way to do it is via the constructor injection?

constructor(@Inject(APP_CONFIG) config: AppConfig

**Reply by timdeschryver** [reply]
@apryk using `inject` also works.

**Comment by selangley-wa** [comment]
FYI. I linked this solution - along with others in this feature request to the Angular Team:

https://github.com/angular/angular/issues/58248
Ability to set environment/configuration values at runtime instead of buildtime

**Comment by smadbe** [comment]
One challenge I faced when implementing this solution is that the configuration is only available where dependency injection is accessible. This creates limitations in places like utility functions and especially NgRx selectors, where DI isn't available. If your application relies on values from the config (such as defaults) in these areas, it can make the transition more complex as it requires to pass a part of the config into the store (the only clean solution I found).

**Reply by timdeschryver** [reply]
You could assign the config to a global variable on `globalThis` or the `window`.

**Reply by smadbe** [reply]
Right. But if we allow the config to be used on `globalThis` or `window`, it becomes difficult to argue for (or even enforce) the use of dependency injection where it's available.
